sycl/include/sycl/queue.hpp

```diff
@@ -2836,32 +2858,35 @@ class __SYCL_EXPORT queue : public detail::OwnerLessBase<queue> {
   /// \param CGF is a function object containing command group.
   /// \param CodeLoc is the code location of the submit call (default argument)
   /// \return a SYCL event object for the submitted command group.
-  template <typename T, typename PropertiesT>
-  std::enable_if_t<std::is_invocable_r_v<void, T, handler &>, event>
-  submit_with_event(
-      PropertiesT Props, T CGF, queue *SecondaryQueuePtr,
+  //
+  // UseFallBackAssert as template param vs `#if` in function body is necessary
+  // to prevent ODR-violation between TUs built with different fallback assert
+  // modes.
```

Comment:
@steffenlarsen , any idea if https://github.com/intel/llvm/blob/a87ac06524c106becfa867965197f192c3e17228/sycl/include/sycl/queue.hpp#L792-L793 is somehow related?

Reply:
The "fallback kernel" in this case is not referring to fallback asserts. It's because the implementation of `ext_oneapi_memcpy2d` has a kernel it uses if the operation isn't natively supported, but we don't want that kernel to pop up unless the user calls `ext_oneapi_memcpy2d` in their code, so we make it dependent.